### PR TITLE
TypeScript rewrite for gh-url-parser with bundler

### PR DESCRIPTION
**Pull request by humphd**
I started to replace the code in the front-end with this new module, but it requires me to do the right thing and properly define all my types.  This converts it to use TS and bundle it properly so we can use it in CommonJS or ESM.

### CODE REVIEW

File: src/github-url-parser/src/extract-github-info.ts [hunk 1]
```diff
@@ -1,15 +1,29 @@
...
+  }
+  return ghUrl;
+};
+
+export const extractGitHubInfo = (gitHubUrls: string[]): GitHubInfo => {
+  const issues: Set<string> = new Set();
+  const pullRequests: Set<string> = new Set();
+  const repos: Set<string> = new Set();
+  const commits: Set<string> = new Set();
+  const users: Set<string> = new Set();
```
Comment (Andrewnt219) [inline src/github-url-parser/src/extract-github-info.ts:17]: nit
```suggestion
  const users = new Set<string>();
```

**Review by Andrewnt219 — APPROVED** [review]
Yes, this is the way. I was wondering why it was in js in the first place, but then I saw the related PR for using in backend.

[tsup](https://github.com/egoist/tsup) is also a great alternative for out of the box TS compilation. Used by [Turborepo](https://github.com/vercel/turborepo/tree/main/examples/design-system).

File: src/github-url-parser/.eslintrc.js [hunk 2]
```diff
@@ -1,10 +1,26 @@
...
+    ecmaVersion: 2021,
+  },
   env: {
-    node: true,
     browser: true,
-    commonjs: true,
+    node: true,
     jest: true,
   },
+  plugins: ['@typescript-eslint'],
```
Comment (DukeManh) [inline src/github-url-parser/.eslintrc.js:12]: The test uses `JS` right? Can we put this in `overrides`
```suggestion
    overrides: [
    {
      parser: '@typescript-eslint/parser',
      files: 'src/**/*.ts',
      plugins: ['@typescript-eslint'],
      rules: {
        '@typescript-eslint/no-unused-vars': 'off',
        '@typescript-eslint/explicit-function-return-type': 'off',
        '@typescript-eslint/explicit-module-boundary-types': 'off',
      },
    },
  ],
```

File: src/github-url-parser/.eslintrc.js [hunk 3]
```diff
@@ -1,10 +1,37 @@
 module.exports = {
   extends: ['@senecacdot/eslint-config-telescope'],
-
+  parser: '@typescript-eslint/parser',
```
Comment (DukeManh) [inline src/github-url-parser/.eslintrc.js:3]:
```suggestion
  parser: '@typescript-eslint/parser',
```

File: src/github-url-parser/.eslintrc.js [hunk 4]
```diff
@@ -1,10 +1,37 @@
...
+        '@typescript-eslint/no-unused-vars': 'off',
+        '@typescript-eslint/explicit-function-return-type': 'off',
+        '@typescript-eslint/explicit-module-boundary-types': 'off',
+      },
+    },
+  ],
+  rules: {
+    '@typescript-eslint/no-unused-vars': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/explicit-module-boundary-types': 'off',
```
Comment (DukeManh) [inline src/github-url-parser/.eslintrc.js:27]:
```suggestion
```

**Review by DukeManh — CHANGES_REQUESTED** [review]

**Comment by humphd** [comment]
Applied your fixes.

**Review by Andrewnt219 — APPROVED** [review]

**Review by DukeManh — APPROVED** [review]

**Review by menghif — CHANGES_REQUESTED** [review]

File: src/github-url-parser/package.json [hunk 5]
```diff
@@ -1,26 +1,34 @@
...
+  "source": "src/index.ts",
+  "main": "dist/main.js",
+  "module": "dist/module.js",
+  "types": "dist/types.d.ts",
   "scripts": {
-    "build": "pnpm types",
-    "test": "jest -c jest.config.js",
+    "pretest": "pnpm build",
+    "test": "run-s build jest",
+    "jest": "jest -c jest.config.js",
```
[the remark this answers is not in the document]
Comment (humphd) [reply src/github-url-parser/package.json:12]: To build the TypeScript into JS we do, yes.  There is probably a clever way to have Jest do this automatically, but I don't know what it is.  If someone wants to do that in a follow-up, they can.

[the remark this answers is not in the document]
Comment (humphd) [reply src/github-url-parser/package.json:12]: @menghif can I merge this?

[the remark this answers is not in the document]
Comment (menghif) [reply src/github-url-parser/package.json:12]: Makes sense! Yes you can.

**Review by menghif — APPROVED** [review]

**Pull request merged**